### ISSUE TITLE
Problems with IRAF in Ubuntu 16.04

### ISSUE THREAD

**Issue by juliotux** [astroconda/astroconda#12]
I'm having some problems to run IRAF from a fresh Astroconda install in Ubuntu 16.04.

When I try to do the ```mkiraf``` command, the following error is returned:
```
Initialize uparm? (y|n): y
-- initializing uparm
Terminal types: xgterm,xterm,gterm,vt640,vt100,etc.
Enter terminal type [default xterm-256color]: xgterm
/home/julio/.anaconda2/bin/mkiraf: 126: /home/julio/.anaconda2/bin/mkiraf: cannot open /unix/hlib/login.cl: No such file
A new LOGIN.CL file has been created in the current directory.
You may wish to review and edit this file to change the defaults.
You may also add additional customizations to loginuser.cl
```

And, when I put my tradicional customized login.cl in the folder, when the `cl` starts, just one line apears, written `curdir:`. If I write the path to the current directory manually, the IRAF stars, but the command return anything. No problem in Fedora was detected.

**Comment by jhunkeler** [comment]
Judging from your path, `/home/julio/.anaconda2/bin/mkiraf`, it looks like you installed IRAF into Anaconda's `root` environment. Our IRAF implementation provides a set of activation and deactivation scripts that **only** work when you install the software into a named environment using the `conda create` command, and use `source activate [envname]` to initialize it.

Here's what to do...

_(Read the next line before clicking this link)_ 
http://astroconda.readthedocs.io/en/latest/faq.html#i-installed-astroconda-into-my-anaconda-root-environment-what-now

**Hint:**
Take a snapshot of your installed packages, stripping any lines containing the word "astroconda", and write them to a file:
```
$ conda list --explicit | sed '/astroconda/d' > conda-root.env
```

After reinstalling Anaconda, use the `conda-root.env` file to restore your original packages:
```
$ conda install --file conda-root.env
```

After restoring your environment, take a look at the following instructions:
http://astroconda.readthedocs.io/en/latest/installation.html#installation-with-legacy-iraf-support

**Comment by juliotux** [comment]
With a separated astroconda env I had the same problems with the `cl` command. The only problem solved is the problem with `mkiraf`.
```
curdir: 
ERROR: Cannot change directory to `'
     called as: `chdir ()'
     called as: `cl ()'
     called as: `cl ()'
```
This problem seems to be related to the one reported here, but no sugestion there solved the problem: http://iraf.net/forum/viewtopic.php?showtopic=1468082

Just for note: In Fedora, no problem was found, even if I run `mkiraf ` with Anaconda root env.

**Comment by jhunkeler** [comment]
@juliotux I posted a workaround in the same thread so others might benefit as well:
http://iraf.net/forum/viewtopic.php?msg=1&showtopic=1468082&lastpost=true#1469408

The workaround involves changing `printf ("!pwd\n") | cl () | scan (curdir)` to become `curdir = osfn("PWD")` instead.

**Comment by juliotux** [comment]
Confirmed this workaround solved the problem.

**Comment by jhunkeler** [comment]
@juliotux After more digging I discovered the `SHELL` variable is not being set to an absolute path under Ubuntu 16.04 (and OS X 10.11). I'm not sure how/why this is happening all of a sudden, but I was able to further workaround the issue by defining a sane value:

## Problem:

```
printenv $SHELL
bash
```

## Solution:
```
export SHELL=/bin/bash
```

The real issue is, `cl` will not be able to execute shell commands until you apply this fix. Even if you're not using the `!cmd` convention on a daily basis, a lot scripts in IRAF rely on it.

**Comment by jhunkeler** [comment]
Well, it turns out this is a bug in Conda's `activate` script, so I opened an issue here:
https://github.com/conda/conda/issues/2979

**Comment by jhunkeler** [comment]
Continuum accepted my proposed change to Conda's `activate` script, however it is not yet available in the latest release (4.0.9). I'll update this issue if/when it's completely fixed upstream.

**Comment by jhunkeler** [comment]
The relase of Conda 4.1.10 fixed this issue. 
`$SHELL` is no longer being clobbered 🎅
